Src/Support/GoogleApis/Apis/[Media]/Upload/ResumableUpload.cs

```diff
@@ -601,22 +580,22 @@ public Task<IUploadProgress> ResumeAsync(Uri uploadUri)
         }
 
         /// <summary>
-        /// Initializes the resumable upload by calling the resumable rest interface to get a unique upload location.
-        /// See https://developers.google.com/drive/manage-uploads#start-resumable for more details.
+        /// Initiates the resumable upload session and returns the session URI.
+        /// See https://developers.google.com/drive/manage-uploads#start-resumable and
+        /// https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload#start-resumable for more information.
         /// </summary>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
         /// <returns>
-        /// The unique upload location for this upload, returned in the Location header
+        /// The task containing the session URI to use for the resumable upload.
         /// </returns>
-        private async Task<Uri> InitializeUpload(CancellationToken cancellationToken)
+        protected virtual Task<Uri> InitiateSessionAsync(CancellationToken cancellationToken)

```

Comment:
An alternative approach might be to make this abstract, and have a subclass (returned from the factory method) that just returned the value? That feels a little cleaner - I *tend* to prefer classes to be either abstract or sealed, although I certainly haven't stamped that preference on this library anywhere.